### PR TITLE
Add host info to start configuration interface

### PR DESCRIPTION
**Pull request by zeusongit**
### Purpose

This PR adds the HostAnalyticsInfo to IStartConfiguration. The HostAnalyticsInfo will be set on host application and will be used at various points, since Application.Current can no longer be used to differentiate between Sandbox and host environment.
This PR covers a single use case that will fix the crash occurring on SplashScreen close in Revit. Other cases where Application.Current is used will also be replaced in a separate PR after this solution is tested in Revit context.

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated
- [ ] This PR contains no files larger than 50 MB

### Release Notes

- Add HostAnalyticsInfo to IStartConfiguration interface.
- Replace Application.Current Check in SplashScreen with HostName check.

### Reviewers

@DynamoDS/dynamo

### CODE REVIEW

File: src/DynamoCore/Models/DynamoModel.cs [hunk 1]
```diff
@@ -535,6 +535,8 @@ public interface IStartConfiguration
             /// Configuration option to start Dynamo in offline mode.
             /// </summary>
             bool NoNetworkMode => false;
+
+            HostAnalyticsInfo HostAnalyticsInfo { get; set; }
```
Comment (QilongTang) [inline src/DynamoCore/Models/DynamoModel.cs:539]: Comments?

**Review by QilongTang — COMMENTED** [review]
Any place else you need to update the usage on `Application.Current` in Dynamo code base?

**Comment by zeusongit** [comment]
> Any place else you need to update the usage on `Application.Current` in Dynamo code base?

There are 12 more places in Dynamo code base where it is used, I was hoping to update those after this change gets in and we test it on Revit locally, or would you prefer including them in this PR itself?

**Comment by QilongTang** [comment]
> > Any place else you need to update the usage on `Application.Current` in Dynamo code base?
> 
> There are 12 more places in Dynamo code base where it is used, I was hoping to update those after this change gets in and we test it on Revit locally, or would you prefer including them in this PR itself?

Oh sure, let's get this one is first since this is API change. As long as the one comment is addressed, I am fine merging.

**Review by QilongTang — APPROVED** [review]

**Review by aparajit-pratap — COMMENTED** [review]
@zeusongit are you going to make the neccessary changes in D4R as well?

**Comment by QilongTang** [comment]
> @zeusongit are you going to make the neccessary changes in D4R as well?

Yes, @zeusongit will propose the PR just need the new Nuget published

**Comment by zeusongit** [comment]
> @zeusongit are you going to make the neccessary changes in D4R as well?

Yes: https://github.com/DynamoDS/DynamoRevit/pull/2999/files

**Pull request merged**